.github/pull_request_template.md

```diff
@@ -0,0 +1,31 @@
+## 📌 Related Issue
+
+<!-- Link to the issue this PR is addressing, e.g., "Closes #42" -->
+
+Closes #
+
+## ✨ What’s been added or changed?
+
+## <!-- Summarize the key changes or features introduced in this PR -->
+
+## 🧪 How was it tested?
+
+## <!-- Describe how you verified the change works (unit tests, notebooks, CLI runs, etc.) -->
+
+## 📸 Screenshots (if applicable)
+
+## <!-- Add before/after screenshots or logs if the changes affect the UI or output -->
+
+## 🧠 Anything to keep in mind?
+
+## <!-- Note anything relevant like known issues, limitations, or design decisions -->
+
+## 🔍 Checklist
+
+<!-- Check off what you've done before submitting the PR -->
+
+- [ ] I've named the title of this pull request starting by 'gh-#: TITLE'
+- [ ] I’ve linked the related issue
+- [ ] I’ve tested the code manually
+- [ ] I’ve added comments/docstrings where needed (i.e: README)
+- [ ] I’ve updated relevant docs if necessary
```

Comment:
I'd also add something like:
`- [ ] There are no breaking changes in the interface/api` 

What I mean here is that no methods are being renamed, no import paths are changing etc. Of course, those cases will happen as we move between different versions, but there should be a period of warning to the deprecation of previous interfaces.

I would add "unit tests"as a check too but I understand in case that's not a priority :D